src/Pim/Component/Api/Normalizer/Exception/ViolationNormalizer.php

```diff
@@ -82,10 +90,19 @@ protected function normalizeViolations(ConstraintViolationListInterface $violati
                 $productValue = $violation->getRoot()->getValues()->getByKey($matches['attribute']);
                 $attributeType = $productValue->getAttribute()->getType();
 
-                $propertyPath = AttributeTypes::IDENTIFIER === $attributeType ? 'identifier' : $violation->getPropertyPath();
+                if (AttributeTypes::IDENTIFIER === $attributeType) {
+                    $propertyPath = 'identifier';
+                }
+
+                $uniqueValueConstraint = new UniqueValue();
```

Comment:
you should put this code in the condition ` if (AttributeTypes::IDENTIFIER === $attributeType) {` 
because if a product value other than identifier is unique, the message will always be `The same identifier is already set on another product`, which is not correct